internal/handler/userHandler.go

```diff
@@ -82,19 +94,29 @@ func (h *Handler) Save(context *gin.Context) {
 // @Description Find user by provided id
 // @Tags Users
 // @Produce json
-// @Param id path int true "User id"
-// @Success 201 {object} response.UserResponseDto "User Found successfully"
-// @Failure 422 {object} response.ErrorDto "Id in the wrong format"
+// @Param id path string true "User id"
+// @Failure 422 {object} response.ErrorDto "ID in the wrong format"
 // @Failure 404 {object} response.ErrorDto "User not found"
+// @Failure 500 {object} response.ErrorDto "Failed to connect to database"
 // @Router /users/{id} [get]
 func (h *Handler) FindById(context *gin.Context) {
 	id := context.Param("id")
 	logger.Infof("Finding user by id %s", id)
 	user, err := h.service.GetUserById(context, id)
 	if err != nil {
+		var databaseConnectionErr *exceptions.DatabaseConnectionErr
+		if errors.As(err, &databaseConnectionErr) {
+			logger.Errorf("Error while connecting to database. %v", err)
+			errorResponse := response.ErrorDto{
+				Message:   "Something went wrong",
+				Timestamp: time.Now(),
+			}
+			context.IndentedJSON(http.StatusInternalServerError, errorResponse)
+			return
+		}
 		logger.Errorf("Error while finding user by id %s. %v", id, err)
 		errorResponse := response.ErrorDto{
-			Message:   err.Error(),
+			Message:   "No user found with id " + id,
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusNotFound, errorResponse)
```

Comment:
Based on my prior comments about the user repository if error is not nil we should log the error and return a 500 error with a generic error message. Since we can't handle any error we got from the repository we should not attempt to check the error type. If the error is nil the enxt step would be to check if the user is nil. If the user is nil we would then return a 404 status.

Reply:
But error is just nil when the result is a success, that's make more sense to work. if there is no user in the database, I should return an error, because it is an error. Not just return nil user and nil error. Mongo returns an error for that too, so makes more sense returning an error